
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>2. Species trees &#8212; Concepts in population genetics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!--
    this give us a css class that will be invisible only if js is disabled
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- So that users can add custom icons -->
  <script src="_static/scripts/fontawesome.js?digest=8878045cc6db502f8baf"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf" />

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'species_trees';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="3. Pedigrees and “gene trees”" href="pedigrees_gene_trees.html" />
    <link rel="prev" title="1. Evolutionary graphs" href="evolutionary_graphs.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
  
    <p class="title logo__title">Concepts in population genetics</p>
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Concepts in population genetics
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Graphs in evolutionary biology</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="evolutionary_graphs.html">1. Evolutionary graphs</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">2. Species trees</a></li>
<li class="toctree-l1"><a class="reference internal" href="pedigrees_gene_trees.html">3. Pedigrees and “gene trees”</a></li>
<li class="toctree-l1"><a class="reference internal" href="pedigrees_random_mating.html">4. Pedigrees in randomly mating populations</a></li>
<li class="toctree-l1"><a class="reference internal" href="mutations_on_gene_trees.html">5. Mutations on gene trees</a></li>
<li class="toctree-l1"><a class="reference internal" href="population_graphs.html">6. Population graphs</a></li>
<li class="toctree-l1"><a class="reference internal" href="gene_trees_to_data.html">7. Gene trees and DNA data</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Population genetic data</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="allele_frequency_spectrum.html">8. The allele frequency spectrum</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="gene_dropping_explained.html">9. Gene dropping on pedigrees</a></li>
<li class="toctree-l1"><a class="reference internal" href="bibliography.html">10. References cited</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
      <div class="sidebar-primary-item">
<div id="ethical-ad-placement"
      class="flat"
      data-ea-publisher="readthedocs"
      data-ea-type="readthedocs-sidebar"
      data-ea-manual="true">
</div></div>
  </div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/molpopgen/popgen_nodes" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/molpopgen/popgen_nodes/issues/new?title=Issue%20on%20page%20%2Fspecies_trees.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/species_trees.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Species trees</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#evolutionary-change-on-trees">2.1. Evolutionary change on trees</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multiple-mutations-at-a-site">2.2. Multiple mutations at a site</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary">2.3. Summary</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="species-trees">
<span id="id1"></span><h1><span class="section-number">2. </span>Species trees<a class="headerlink" href="#species-trees" title="Link to this heading">#</a></h1>
<p>The following image shows the relationship of three species – human, chimpanzee, and gorilla.
The y axis depicts time in the <em>past</em>, which is increasing from the bottom to the top of the image.
The time units are “millions of years ago”, or <code class="docutils literal notranslate"><span class="pre">mya</span></code>.</p>
<figure class="align-default" id="human-chimp-gorilla">
<img alt="_images/human_chimp_gorilla.svg" src="_images/human_chimp_gorilla.svg" />
<figcaption>
<p><span class="caption-number">Fig. 2.1 </span><span class="caption-text">A “species tree” showing the relationship between three species: human, chimpanzee, and gorilla.</span><a class="headerlink" href="#human-chimp-gorilla" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>The image contains the following elements:</p>
<ol class="arabic simple">
<li><p><em>nodes</em>: a node is a vertex on the graph that represents an ancestor or a terminal “tip”.
On this tree, tips are shown as squares and branching points are circles.
This graph has three terminal tips, which are our modern day species (human/chimp/gorilla).
This graph has two ancestor nodes, which are associated with <em>branching points</em>, meaning that there is more than one line moving from the node towards the present time.
On thise tree, these branching point nodes are the <em>common ancestors</em> of our species.
The human/chimpanzee common ancestor is labelled <code class="docutils literal notranslate"><span class="pre">H/C</span></code> and the common ancestor of <code class="docutils literal notranslate"><span class="pre">H/C</span></code> and gorilla
is labelled <code class="docutils literal notranslate"><span class="pre">H/C/G</span></code>.</p></li>
<li><p><em>edges</em>: edges are the lines connecting the nodes.  In this image, the y axis represents time.
Thus, the lengths of the edges are in units of <em>millions of years</em>.</p></li>
</ol>
<p>A set of <em>nodes</em> connected by <em>edges</em> is a <em>graph</em>. A graph with no cycles (nodes connected by edges to form “loops”) is a <em>tree</em>.</p>
<p>In most use cases in evolutionary biology, we call an <em>edge</em> a <em>branch</em> because we call the graph a tree!!</p>
<p>Further, because the graph is oriented with respect to <em>time in the past</em>, we can say things like the following:</p>
<ol class="arabic simple">
<li><p>The human/chimp common ancestor existed 6 million years ago.  We get this simply from looking at our image. (Although we’d usually get this information from some file that we use to plot the tree.)</p></li>
<li><p>The human/chimp/gorilla common ancestor existing 12 million years ago.</p></li>
<li><p>Human and chimps are more closely related to each other than either one is to the gorilla.  We make this statement because:</p>
<ol class="arabic simple">
<li><p>The human/chimp common ancestor occurred in the more recent past than the common ancestor or human/gorilla (which is the same node as the chimp/gorilla common ancestor.)</p></li>
<li><p>Therefore, there is more <em>time during which evolution (change) occurred</em> between human/gorilla (and also chimp/gorilla) than there is between human and chimp.</p></li>
</ol>
</li>
</ol>
<p>To illustrate this final point in more detail, let us show the same tree with each branch given a different color:</p>
<figure class="align-default" id="id2">
<img alt="_images/human_chimp_gorilla_colored_edges.svg" src="_images/human_chimp_gorilla_colored_edges.svg" />
<figcaption>
<p><span class="caption-number">Fig. 2.2 </span><span class="caption-text">The same species tree as <a class="reference internal" href="#human-chimp-gorilla"><span class="std std-numref">Fig. 2.1</span></a> with the branches now colored.</span><a class="headerlink" href="#id2" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>We have the following branches and <em>branch lengths</em>:</p>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>Ancestor (parent)</p></th>
<th class="head"><p>Descendant (child)</p></th>
<th class="head"><p>Color</p></th>
<th class="head"><p>Length (mya)</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>H/C/G</p></td>
<td><p>H/C</p></td>
<td><p>Pink</p></td>
<td><p>6</p></td>
</tr>
<tr class="row-odd"><td><p>H/C/G</p></td>
<td><p>Gorilla</p></td>
<td><p>Purple</p></td>
<td><p>12</p></td>
</tr>
<tr class="row-even"><td><p>H/C</p></td>
<td><p>Human</p></td>
<td><p>Cyan</p></td>
<td><p>6</p></td>
</tr>
<tr class="row-odd"><td><p>H/C</p></td>
<td><p>Chimp</p></td>
<td><p>Red</p></td>
<td><p>6</p></td>
</tr>
</tbody>
</table>
</div>
<p>We get the branch lengths directly from the tree.  For example, human exists now (0 mya) and the <code class="docutils literal notranslate"><span class="pre">H/C</span></code> node is 6mya ago. The branch length is therefore <span class="math notranslate nohighlight">\(6-0=6\)</span>.</p>
<p>Evolutionary change accumulates over time.
This accumulation is the result of mutations that mutations that affect organismal phenotypes.
In order to become species differences, the mutations have to “stick around” long enough to spread through entire species.</p>
<p>We can use our branch lengths to ask how much evolutionary time has occurred between our three “tip” species.
For example:</p>
<ol class="arabic simple">
<li><p>6 million years separate human from the <code class="docutils literal notranslate"><span class="pre">H/C</span></code> node.</p></li>
<li><p>6 million years separate chimp from the <code class="docutils literal notranslate"><span class="pre">H/C</span></code> node.</p></li>
<li><p>Therefore <span class="math notranslate nohighlight">\(6 + 6 = 12\)</span> million years separate modern humans from contemporary chimpanzees.</p></li>
</ol>
<p>This simple calculation illustrates a general principle:</p>
<p>The amount of evolution that has occurred between two species on a tree is a function of the <strong>sum</strong> of the branch lengths separating them.
We say “is a function of” because we also need to consider how quickly phenotypes or DNA sequences change per unit time.</p>
<section id="evolutionary-change-on-trees">
<h2><span class="section-number">2.1. </span>Evolutionary change on trees<a class="headerlink" href="#evolutionary-change-on-trees" title="Link to this heading">#</a></h2>
<p>Let us expand on our previous discussion, using DNA changes as a concrete example.
The following graphic shows our tree with three mutations added.
The notation for a mutation is: <code class="docutils literal notranslate"><span class="pre">Position:</span> <span class="pre">ancestral</span> <span class="pre">state</span> <span class="pre">-&gt;</span> <span class="pre">derived</span> <span class="pre">state</span></code>.
In words, we read these labels as, “At Position X in the genome, the DNA sequence changed from <code class="docutils literal notranslate"><span class="pre">ancestral</span> <span class="pre">state</span></code> to <code class="docutils literal notranslate"><span class="pre">derived</span> <span class="pre">state</span></code>.”</p>
<p>The ancestral state is usually the DNA state that was present in the node that is ancestral to (more ancient than) the mutation.
The positions of each mutation on a branch represent when they occurred in the past.
For example, at position 800, the <code class="docutils literal notranslate"><span class="pre">H/C/G</span></code> node has a <code class="docutils literal notranslate"><span class="pre">T</span></code>.
At about 11mya, this position changed from <code class="docutils literal notranslate"><span class="pre">T</span></code> to <code class="docutils literal notranslate"><span class="pre">A</span></code> on the branch leading from <code class="docutils literal notranslate"><span class="pre">H/C/G</span></code> to Gorilla.
The result is that most or all individuals that are modern gorillas will have an <code class="docutils literal notranslate"><span class="pre">A</span></code> at position 800.</p>
<p>We can write down the logic required to figure out how to determine the DNA sequence at a specific position for our tip nodes:</p>
<ol class="arabic simple">
<li><p>Identify the <code class="docutils literal notranslate"><span class="pre">derived</span></code> state of the change. (<code class="docutils literal notranslate"><span class="pre">A</span></code> at position 800.)</p></li>
<li><p>Identify the <code class="docutils literal notranslate"><span class="pre">branch</span></code> that the mutation occurs on. (The <code class="docutils literal notranslate"><span class="pre">H/C/G</span></code> to Gorilla branch for this mutation.)</p></li>
<li><p>Trace from the mutation down to each tree tip, making sure to follow all branching points along the way. This step is easy for position 800 – there is only one branch all the way down to the Gorilla tip node.  The derived state is present in all of the tips.</p></li>
</ol>
<p>Position 800 is not very interesting, but position 42 is!
Let’s perform our steps:</p>
<ol class="arabic simple">
<li><p>The derived state of this mutation is <code class="docutils literal notranslate"><span class="pre">C</span></code>.</p></li>
<li><p>It occurs on the <code class="docutils literal notranslate"><span class="pre">H/C/G</span></code> to <code class="docutils literal notranslate"><span class="pre">H/C</span></code> branch.</p></li>
<li><p>Tracing down the tree to the present day, we find that the tips are Human and Chimp. Therefore, human and chimp <em>both</em> have a <code class="docutils literal notranslate"><span class="pre">C</span></code> at position 42 of the genome!</p></li>
</ol>
<figure class="align-default" id="human-chimp-gorilla-with-mutations">
<img alt="_images/human_chimp_gorilla_with_mutations.svg" src="_images/human_chimp_gorilla_with_mutations.svg" />
<figcaption>
<p><span class="caption-number">Fig. 2.3 </span><span class="caption-text">Adding mutations to our species tree.</span><a class="headerlink" href="#human-chimp-gorilla-with-mutations" title="Link to this image">#</a></p>
</figcaption>
</figure>
</section>
<section id="multiple-mutations-at-a-site">
<span id="multiple-hits"></span><h2><span class="section-number">2.2. </span>Multiple mutations at a site<a class="headerlink" href="#multiple-mutations-at-a-site" title="Link to this heading">#</a></h2>
<p>Unfortunately, biology is quite messy.
Above, we noted that the ancestral state is “usually” the one found at the ancestor node.
We had to say “usually” because the same site in a genome can change multiple times.
This occurs when there is a lot of evolutionary time on a branch.
(In other words, when it is long.)
Mutations happen every generation, and over long time periods, the same site can be “hit” more than once by a mutation.</p>
<p>The following graphic illustrates such “multiple hits”:</p>
<figure class="align-default" id="human-chimp-gorilla-with-multiple-hits">
<img alt="_images/human_chimp_gorilla_with_multiple_hits.svg" src="_images/human_chimp_gorilla_with_multiple_hits.svg" />
<figcaption>
<p><span class="caption-number">Fig. 2.4 </span><span class="caption-text">A species tree where multiple mutations have occurred at the same position during evolution.</span><a class="headerlink" href="#human-chimp-gorilla-with-multiple-hits" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>We now have additional mutations that occurred at positions 42 and 800.
When mutations occur at the same position, the <em>ancestral</em> state is the <em>derived</em> state of the previous mutation at that position.  For example, at position 42, we have a mutation from <code class="docutils literal notranslate"><span class="pre">T</span></code> to <code class="docutils literal notranslate"><span class="pre">C</span></code> about 7 million years ago and then another from <code class="docutils literal notranslate"><span class="pre">C</span></code> to <code class="docutils literal notranslate"><span class="pre">T</span></code> about 0.5 million years ago on the branch leading to chimpanzees.  The result of this chain of events is that humans have a <code class="docutils literal notranslate"><span class="pre">C</span></code> at this position yet the chimp has a <code class="docutils literal notranslate"><span class="pre">T</span></code> because the second “hit” reverted the DNA sequence back to the original ancestral state (<code class="docutils literal notranslate"><span class="pre">T</span></code>).</p>
<p>Similarly, a second mutation at position 800 on the gorilla branch leads to a final state of <code class="docutils literal notranslate"><span class="pre">C</span></code> at position 800.</p>
<p>Multiple mutations at the same position mean that we need to update our logic.
As we trace a mutation at a site, we start with the most ancient mutation.
For each mutation we encounter, we record its derived state.
If we encounter another mutation at the same state, we replace the previous derived state with the value from the more recent mutation.
We have to take care to recognize when different branches can have different derived states at a given site.  Position 42 is an example of this phenomenon because the most recent mutation is more recent than the split of humans and chimpanzees.</p>
</section>
<section id="summary">
<h2><span class="section-number">2.3. </span>Summary<a class="headerlink" href="#summary" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p>Trees are graphs that describe the relationship among the tip nodes.</p></li>
<li><p>In biology, the tips of trees represent things that we study – extant species, individuals sampled from modern populations, etc..</p>
<ol class="arabic simple">
<li><p>Tips need not be associated with the present time. We are now able to sequence DNA from preserved samples.  For example, we have genome sequences from extinct species like the wooly mammoth and the Neanderthal.</p></li>
</ol>
</li>
<li><p>The ancestral (non-tip) nodes are the ancestors of the tips, meaning that they existed in the past.</p></li>
<li><p>Branch lengths usually indicate “how much evolution” has occurred on the branch. The examples in this section are in units of time.</p></li>
<li><p>We can trace changes on the trees from the past to the present.  These changes tell us the “state” of the tips. The examples here all trace changes in DNA sequence from the past to the present.</p></li>
</ol>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="evolutionary_graphs.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title"><span class="section-number">1. </span>Evolutionary graphs</p>
      </div>
    </a>
    <a class="right-next"
       href="pedigrees_gene_trees.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">3. </span>Pedigrees and “gene trees”</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#evolutionary-change-on-trees">2.1. Evolutionary change on trees</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multiple-mutations-at-a-site">2.2. Multiple mutations at a site</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary">2.3. Summary</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Kevin R. Thornton
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf"></script>
<script defer src="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>